service/storage_service.cc

```diff
@@ -205,12 +205,36 @@ bool get_property_load_ring_state() {
     return get_local_storage_service().db().local().get_config().load_ring_state();
 }
 
-bool storage_service::should_bootstrap() const {
-    return is_auto_bootstrap() && !db::system_keyspace::bootstrap_complete() && !_gossiper.get_seeds().count(get_broadcast_address());
+bool storage_service::is_first_node() {
+    if (db().local().is_replacing()) {
+        return false;
+    }
+    auto seeds = _gossiper.get_seeds();
+    if (seeds.empty()) {
+        return false;
+    }
+    // Node with the smallest IP address is chosen as the very first node
+    // in the cluster. The first node is the only node that does not
+    // bootstrap in the cluser. All other nodes will bootstrap.
```

Comment:
Good. Since it's illegal to bootstrap a cluster with different versions we won't run into different decisions by different nodes.

The only issue is if different nodes have different seed lists. We must document against it.

Reply:
Yes. We need to document and enforce this.